internal/store/builder_test.go

```diff
@@ -113,3 +113,86 @@ func TestWithAllowLabels(t *testing.T) {
 		}
 	}
 }
+
+func TestWithAllowAnnotations(t *testing.T) {
+	tests := []struct {
+		Desc                 string
+		AnnotationsAllowlist map[string][]string
+		EnabledResources     []string
+		Wanted               LabelsAllowList
```

Comment:
should this be
```suggestion
		Wanted               AnnotationsAllowList
```

?

Reply:
Well, It reuses the same type from the test above [here](https://github.com/kubernetes/kube-state-metrics/pull/2234/files/e4d2bd48c8cfa950941dd51149a75ffbbb8f44eb#diff-c8cf0574c6fe66c4a3d8522108993b75d0421ec94af5eab80c3ed25d01d64cd1R26). I could declare a new type and point it to `options.LabelsAllowList`, while at it should I change [this](https://github.com/kubernetes/kube-state-metrics/blob/82fd15ccc4a4c716721fcb135b1f7ef129025a3f/pkg/options/options.go#L32) to a more generic also? 

I noticed that these two tests test the same function in the end, I wrote tests first then went to implement, remove the test or keep it?

Reply:
Ah I didn't know AnnotatiosAllowList don't exist. I think we can keep it for this change like this. :)